.github/workflows/monitor.yml

```diff
@@ -0,0 +1,44 @@
+name: Root Monitor
+
+# Execute this as a daily cron job.
+on:
+  # Enable cron for re-signing snapshot and timestamp every week
+  schedule:
+    - cron: '0 0 */1 * *' # every day
+  workflow_dispatch:
+
+jobs:
+  root-probe:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.7
+      - name: Install verify
+        run: go install github.com/sigstore/root-signing/...@latest
+
+      # Check expiration in less than 3 days
+      - name: Check expiration
+        run: |
+          export EXPIRY=$(date -d '+3 days' '+%Y/%m/%d')
+          verify repository --gcs --repository sigstore-tuf-root --root ceremony/2021-06-18/repository/root.json --valid-until ${EXPIRY}
+
+  slack-notification:
+    name: Slack Notification On Failure
+    runs-on: ubuntu-latest
+
+    needs: [root-probe]
+    if: always() && (needs.root-probe.result == 'failure')
+
+    steps:
+    - name: Set message
+      run: |
+        echo "root_state=failure" >> $GITHUB_ENV
```

Comment:
You can pass values between steps in the same job with [`::set-output`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter), which feels a bit cleaner?

As it stands, I'd be inclined to remove this step and hard-code the error message (as we currently only report on failure).